### PR TITLE
[TECH-280] MultiLevel Reseller Detection Fix and Update

### PR DESCRIPTION
**Pull request by cPanelRicky**
Fixes [TECH-280] - Zamfoo detection no longer works

- Change Zamfoo plugin check to look for plugin directory rather than checking for the word Zamfoo in /scripts/prekillacct
- Consolidate all Multilevel reseller plugin checks into a single check
- Update version to 4.99.156

### CODE REVIEW

**Review by cPanelChrisD — APPROVED** [review]
Looks good!

File: ssp [hunk 1]
```diff
@@ -8285,44 +8282,15 @@ sub check_for_newrelic {
 
 }
 
-sub check_for_zamfoo {
-    my $has_zamfoo = 0;
+sub check_for_multilevel_reseller {
+    my @ml_plugins = ( 'zamfoo', 'whmreseller', 'whmphp', 'whmamp' );
```
Comment (davelcpanelnet-zz) [inline ssp:8286]: I would like to see some clearer standardized array creation when it's static like this. qw() is really the preferred best practice.

EX:
```
• [root@davel:/usr/local/cpanel] (hb_integration_66)# /usr/local/cpanel/3rdparty/bin/perl -MData::Dumper -e 'my @ml_plugins = qw/ zamfoo whmreseller whmphp whmamp /; print Dumper \@ml_plugins'
$VAR1 = [
          'zamfoo',
          'whmreseller',
          'whmphp',
          'whmamp'
        ];
```
This isn't something I am going to reject - but it's best practice.

File: ssp [hunk 2]
```diff
@@ -8285,44 +8282,15 @@ sub check_for_newrelic {
...
+    my $docroot = '/usr/local/cpanel/whostmgr/docroot/cgi';
```
Comment (davelcpanelnet-zz) [inline ssp:8287]: I'd like a different variable name here.  $docroot implies an absolute root RE: Apache so that's a bit misleading.

Comment (davelcpanelnet-zz) [reply ssp:8287]: $cgi_root would work

File: ssp [hunk 3]
```diff
@@ -8285,44 +8282,15 @@ sub check_for_newrelic {
...
+    foreach my $plugin (@ml_plugins) {
+        if ( -d "$docroot/$plugin" ) {
+            print_3rdp( uc($plugin) . ' ' );
+            print_3rdp2('is installed. Multi-level reseller setups are not supported!');
```
Comment (davelcpanelnet-zz) [inline ssp:8291]: Please remove the exclamation

**Review by davelcpanelnet-zz — CHANGES_REQUESTED** [review]
There are some minor changes I would like toe here.
- Array Definition
- Variable name

**Comment by cPanelRicky** [comment]
Instead of closing these when minor changes are needed, if you leave the pull request open the subsequent commits to the branch will automatically be reflected in the pull request removing the need to open a whole new one.

**Comment by davelcpanelnet-zz** [comment]
I will take that into consideration

**Pull request closed without merging**